aws/resource_aws_cloudwatch_event_archive.go

```diff
@@ -0,0 +1,200 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	events "github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsCloudWatchEventArchive() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudWatchEventArchiveCreate,
+		Read:   resourceAwsCloudWatchEventArchiveRead,
+		Update: resourceAwsCloudWatchEventArchiveUpdate,
+		Delete: resourceAwsCloudWatchEventArchiveDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCloudWatchEventArchiveName,
+			},
+			"event_source_arn": {
```

Comment:
This should probably be `ForceNew` since it can't be updated, unless it is expected to be changing on the backend and you don't want to trigger the creation of a new resource.

Reply:
Indeed, great catch.